backend/src/bundles/payment/payment.service.ts

```diff
@@ -119,18 +119,66 @@ class PaymentService implements IPaymentService {
         const subscription: Stripe.Subscription =
             await this.createStripeSubscription(customer.id, priceId);
 
-        const { latest_invoice, id } = subscription;
+        const { latest_invoice, id, status } = subscription;
 
         if (!latest_invoice || typeof latest_invoice === 'string') {
             return {
                 clientSecret: null,
                 subscriptionId: id,
             };
         }
-
-        const { client_secret } =
+        const { client_secret, payment_method } =
             latest_invoice.payment_intent as Stripe.PaymentIntent;
 
+        const retrievedSubscription = await this.stripe.subscriptions.retrieve(
+            id,
+        );
+        const { current_period_start, current_period_end, items } =
```

Comment:
Names of variables should be written in camelCase notation.
Check it in other places.

Reply:
from stripe